src/client.rs

```diff
@@ -22,11 +23,17 @@ impl Plugin for ClientPlugin {
             .add_state::<ClientState>()
             .init_resource::<LastTick>()
             .init_resource::<NetworkEntityMap>()
+            .add_systems(
+                (
+                    Self::no_connection_state_system.run_if(resource_removed::<RenetClient>()),
+                    Self::connecting_state_system
+                        .run_if(bevy_renet::client_connecting)
+                        .run_if(state_exists_and_equals(ClientState::NoConnection)),
+                )
+                    .before(run_enter_schedule::<ClientState>)
+                    .in_base_set(CoreSet::StateTransitions),
+            )
             .add_systems((
-                Self::no_connection_state_system.run_if(resource_removed::<RenetClient>()),
-                Self::connecting_state_system
-                    .run_if(bevy_renet::client_connecting)
-                    .in_set(OnUpdate(ClientState::NoConnection)),
                 Self::connected_state_system
```

Comment:
Could you please put this system to `CoreSet::StateTransitions` too?
```suggestion
                Self::connected_state_system
```